src/tests/ftest/control/version.py

```diff
@@ -22,32 +22,69 @@ class DAOSVersion(TestWithServers):
     def __init__(self, *args, **kwargs):
         """Initialize a DAOSVersion object."""
         super().__init__(*args, **kwargs)
+
         # Don't waste time starting servers and agents.
         self.setup_start_servers = False
         self.setup_start_agents = False
 
+        self.errors = []
+
+    def report_result(self):
+        """Helper printing test result"""
+        self.log.info("###### Test Result ######")
+        report_errors(test=self, errors=self.errors)
+        self.log.info("#########################")
+
+    def append_error(self, title, details):
+        """Helper adding an error to the list of errors
+
+        Args:
+            title (str): Error message title
+            details (list): List of string of the error details
+        """
+        msg = title
+        if details:
+            msg += "\n\t" + "\n\t".join(details)
+        self.errors.append(msg)
+
     def test_version(self):
         """Verify version number for dmg, daos, daos_server, and daos_agent against RPM.
 
         :avocado: tags=all,full_regression
         :avocado: tags=vm
         :avocado: tags=control,daos_cmd
-        :avocado: tags=version_number,test_version
+        :avocado: tags=DAOSVersion,test_version
         """
-        errors = []
-
         # Get RPM version.
         rpm_command = "rpm -qa|grep daos-server"
         output = run_pcmd(hosts=self.hostlist_servers, command=rpm_command)
-        self.log.info("RPM output = %s", output)
-        stdout = output[0]["stdout"][0]
-        self.log.info("RPM stdout = %s", stdout)
-        result = re.findall(r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", stdout)
+        self.log.debug("RPM output = %s", output)
+        rc = output[0]["exit_status"]
+        stdout = output[0]["stdout"]
+        if rc != 0:
+            msg = "DAOS RPMs not properly installed: rc={}".format(rc)
+            self.append_error(msg, stdout)
+            self.report_result()
+        rpm_version = None
+        for rpm in stdout:
+            result = re.findall(r"daos-server-[tests-|tests_openmpi-]*([\d.]+)", rpm)
+            if result:
+                rpm_version = result[0]
+                break
         if not result:
-            errors.append("RPM version is not in the output! {}".format(output))
-        else:
-            rpm_version = result[0]
-            self.log.info("RPM version = %s", rpm_version)
+            msg = "RPM version could not be defined"
+            self.append_error(msg, stdout)
+            self.report_result()
+        self.log.info("RPM version = %s", rpm_version)
+
+        # Remove configuration files
+        cleanup_cmds = [
+            "sudo find /etc/daos/certs -type f -delete -print",
+            "sudo rm -fv /etc/daos/daos_server.yml /etc/daos/daos_control.yml"
+            "     /etc/daos/daos_agent.yml",
+        ]
+        for cmd in cleanup_cmds:
+            run_pcmd(hosts=self.hostlist_servers, command=cmd)
```

Comment:
I will add these config parameters in a follow up PR as this one has been already merged.
Otherwise, I would prefer to keep the cleanup as it seems that the _/etc/daos_ is not properly cleanned between test runs.

- [ ] Add insecure yaml config